### ISSUE TITLE
Feature request: support for independent data

### ISSUE THREAD

**Issue by blankclemens** [danvk/dygraphs#636]
Right now, combining two independent data sets in one graph gives quite some headache; support of independent data would solve this issue immediately.

Let me explain:

Given two data sets

    [ [0, 2], [1, 3], [2, 4] ]
    [ [0.3, 2], [1.76, 3], [2.23, 4] ]

one would have to create a new data set with 6 entries, but then take care of what happens with the other data set at value x.

Do I calculate an interpolation? Do I set them to 'NaN'? What happens then with line connections and error bars? What happens with 'NaN' as the first point? Plus I have to take care at which point exactly the added points have to be, since dygraphs doesn't support unsorted data as well (another feature which would be nice to have).

Quite some questions to take care for, quite some line of codes and calculation for a task, which seems pretty common and simple at first sight. 

What do you guys think?

**Comment by danvk** [comment]
Did you run across [this][1]?

This is a fairly common request, so it might make sense to support independent series in a less hacky way.

[1]: http://dygraphs.com/tests/independent-series.html

**Comment by blankclemens** [comment]
Yes, but this shows exactly the problem: They are not really independet series, instead there's always a null/NaN inserted.

If one has a "clean" x-axis, with fixed measurement points this might be fine, but I use it for measurements in physics where no point has exactly the same time (x-axis) as another. Each added measurement easely doubles the size of the dygraphs data then (since I always have to add null/NaN(*)) plus I have to take care of the questions mentioned above.

(*) Just found out as written on your link that null and NaN behave differently. Is this a wanted behaviour? It appears a bit confusing to me.

**Comment by danvk** [comment]
It's definitely subtle. There's also the `connectSeparatedPoints` option to consider. I'm open to suggestions, though changes at this point would be quite disruptive to existing users.

**Comment by lBowlin** [comment]
I have encountered this on a couple of my projects as well. I was forced to
re-sample the data on the server and interpolate, so that it didn't limit
the size/# of points I could pipe to dygraphs. Having to expand the data in
dygraphs to a sparse matrix doesn't pan out well for large datasets. The
server resample worked well for me, but does slow down every request.
Similarly I could have the client interpolate in javascript, but that has
it's own issues.


On Mon, Jun 29, 2015 at 10:41 AM, Dan Vanderkam <notifications@github.com>
wrote:

> It's definitely subtle. There's also the connectSeparatedPoints option to
> consider. I'm open to suggestions, though changes at this point would be
> quite disruptive to existing users.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danvk/dygraphs/issues/636#issuecomment-116772004>.
>



-- 
*Lars Bowlin*

**Comment by blankclemens** [comment]
Exactly my thoughts.. to lower the calculation power needed for the client I'm thinking about doing it on server side, although this comes with a not too small downside of increasing the data, which needs to be send, by a huge amount.

It might be difficult to implement without breaking backwards compatibility, although it might be able to set it as an option to enter multiple data arrays instead of one for all. If not, it might be interesting for a major update? ;)

**Comment by Dreiffert** [comment]
Hi all,

are there any plans to implement this, at all or anytime soon?

Greetings and a huge thanks to ll the contributers!

**Comment by kierson** [comment]
I am also voting to handle this.
It might also be useful when you implement an app where user can select different data sources (even tough they have the same X values), because now you have to combine them into one data set manually and take care about inserting nulls.
